include/seqan3/io/sequence_file/format_genbank.hpp

```diff
@@ -0,0 +1,252 @@
+     * \{
+     */
+     //!\brief Default constructor is explicitly defaulted, you need to provide a stream or file name.
+    sequence_file_format_genbank() = default;
+    //!\brief Copy construction is explicitly deleted, because you can't have multiple access to the genbank file.
+    sequence_file_format_genbank(sequence_file_format_genbank const &) = delete;
+    //!\brief Copy assignment is explicitly deleted, because you can't have multiple access to the genbank file.
+    sequence_file_format_genbank & operator=(sequence_file_format_genbank const &) = delete;
+    //!\brief Move construction is defaulted.
+    sequence_file_format_genbank(sequence_file_format_genbank &&) = default;
+    //!\brief Move assignment is defaulted.
+    sequence_file_format_genbank & operator=(sequence_file_format_genbank &&) = default;
+    //!\}
+
+    //!\brief The valid file extensions for this format; note that you can modify this value.
+    static inline std::vector<std::string> file_extensions
+    {
+        { "genbank" },
+        { "gb" },
+        { "gbk" },
+    };
+
+    //!\copydoc SequenceFileInputFormat::read
+    template <typename stream_type,     // constraints checked by file
+              typename seq_legal_alph_type, bool seq_qual_combined,
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type>
+    void read(stream_type                                                            	& stream,
+              sequence_file_input_options<seq_legal_alph_type, seq_qual_combined> const & options,
+              seq_type                                                                  & sequence,
+              id_type                                                                   & id,
+              qual_type                                                                 & SEQAN3_DOXYGEN_ONLY(qualities))
+    {
+        using stream_char_t = typename stream_type::char_type;
+        auto stream_view = std::ranges::subrange<decltype(std::istreambuf_iterator<stream_char_t>{stream}),
+                                          decltype(std::istreambuf_iterator<stream_char_t>{})>
+                            {std::istreambuf_iterator<stream_char_t>{stream},
+                             std::istreambuf_iterator<stream_char_t>{}};
+        auto stream_it = std::ranges::begin(stream_view);
+
+        if (!(std::ranges::equal(stream_view | view::take_until_or_throw(is_cntrl || is_blank), std::string{"LOCUS"})))
+            throw parse_error{"An entry has to start with the code word LOCUS."};
+
+        //ID
+        if constexpr (!detail::decays_to_ignore_v<id_type>)
+        {
+            if (options.embl_genbank_complete_header)
+            {
+                std::ranges::copy(std::string{"LOCUS"}, std::back_inserter(id));
+
+                while (!(std::ranges::equal(stream_view | view::take_until_or_throw(is_cntrl || is_blank),
```

Comment:
I think this may fail if the header as another keyword staring with `O` like `ORGANISM`..

Can you test the following header when reading the complete header:
```
LOCUS       SCU49845     5028 bp    DNA             PLN       21-JUN-1999
DEFINITION  Saccharomyces cerevisiae TCP1-beta gene, partial cds, and Axl2p
            (AXL2) and Rev7p (REV7) genes, complete cds.
ACCESSION   U49845
VERSION     U49845.1  GI:1293613
KEYWORDS    .
SOURCE      Saccharomyces cerevisiae (baker's yeast)
  ORGANISM  Saccharomyces cerevisiae
            Eukaryota; Fungi; Ascomycota; Saccharomycotina; Saccharomycetes;
            Saccharomycetales; Saccharomycetaceae; Saccharomyces.
REFERENCE   1  (bases 1 to 5028)
FEATURES             Location/Qualifiers
     source          1..5028
                     /organism="Saccharomyces cerevisiae"
                     /db_xref="taxon:4932"
                     /chromosome="IX"
                     /map="9"
     CDS             <1..206
                     /codon_start=3
                     /product="TCP1-beta"
                     /protein_id="AAA98665.1"
                     /db_xref="GI:1293614"
                     /translation="SSIYNGISTSGLDLNNGTIADMRQLGIVESYKLKRAVVSSASEA
                     AEVLLRVDNIIRARPRTANRQHM"
     gene            687..3158
                     /gene="AXL2"
ORIGIN
        1 gatcctccat atacaacggt atctccacct caggtttaga tctcaacaac ggaaccattg
       61 ccgacatgag acagttaggt atcgtcgaga gttacaagct aaaacgagca gtagtcagct
```

Reply:
It still reads the complete header, I added a more complex header to id1 now.